kolibri/plugins/learn/assets/src/state/actions/main.js

```diff
@@ -237,6 +237,17 @@ function showTopicsTopic(store, id, isRoot = false) {
         topic: _topicState(topic, ancestors),
         contents: topicContents,
       };
+      pageState.channel = currentChannel;
+      if (isRoot) {
+        pageState.topic = _topicState(
+          Object.assign(topic, {
+            description: currentChannel.description,
+          }),
+          ancestors
+        );
+      } else {
+        pageState.topic = _topicState(topic, ancestors);
+      }
```

Comment:
This is good, but it's written in a somewhat confusing way. In particular, the `pageState.topic` assignment can be outside of the if statement since it's common to both branches. In general, you want to limit the contents of code branches to only that code which is different between the cases.

Something like this might work?

```javascript
pageState.channel = currentChannel;
if (isRoot) {
  topic.description = currentChannel.description;
}
pageState.topic = _topicState(topic, ancestors);
```

Reply:
Mea culpa, I forgot that we dropped the 'don't modify function parameters linting rule' when I advised @GCodeON of this approach.